Add support for custom schema prototype

cschiewek opened the pull request:
Currently, the only way to add custom schema directives in Absinthe is to define a schema prototype.  However, absinthe only supports a single prototype, so using schema directives from this library, and your own custom schema prototype, isn't possible.

This PR attempts to solve that by:
- Adding `:skip_prototype` option to `Absinthe.Federation.Schema`
- Bypasses the `@prototype_schema` module attribute when said option is true
- Moves all the directives (and associated types) to a new module, so that they can be included in both the Absinthe Federations default prototype or any dependants prototype.

I also added a test and updated the README with instructions on how to configure when downstream projects already have a prototype.

Please let me know if there's an alternative approach you'd prefer.

kzlsakal commented:
@cschiewek nice! I was thinking about using Absinthe's `import_directives` macro (which was made available with 1.7.1) in the future. This is a great workaround for the current implementation.

kzlsakal commented on README.md:
````diff
@@ -42,6 +42,31 @@ defmodule MyApp.MySchema do
 end
 ```
 
+If you are already using a schema prototype
````
What do you think about moving this under the "Usage" right below "Macro based schemas" example?

cschiewek replied:
Will do!

kzlsakal commented on lib/absinthe/federation/schema/prototype/directives.ex:
```diff
@@ -0,0 +1,146 @@
+defmodule Absinthe.Federation.Schema.Prototype.Directives do
```
I think we could rename this to `Absinthe.Federation.Schema.FederatedDirectives`, what do you think?

cschiewek replied:
Agreed.  I'll update shortly.

cschiewek commented:
> @cschiewek nice! I was thinking about using Absinthe's `import_directives` macro (which was made available with 1.7.1) in the future. This is a great workaround for the current implementation.

I tried that first, `import_directives` only works with operation directives, not schema directives, so you can't use them.

kzlsakal commented:
> > @cschiewek nice! I was thinking about using Absinthe's `import_directives` macro (which was made available with 1.7.1) in the future. This is a great workaround for the current implementation.
> 
> I tried that first, `import_directives` only works with operation directives, not schema directives, so you can't use them in a prototype module.

Good to know, and that's a bummer :D

cschiewek commented:
@kzlsakal I've addressed all the feedback (TY!) and also added descriptions for the `_service` and `sdl` fields, as apollo's linter complained that they were missing.

cschiewek commented on lib/absinthe/federation/types.ex:
```diff
@@ -9,7 +9,16 @@ defmodule Absinthe.Federation.Types do
+    @desc """
+    This SDL (schema definition language) is a printed version of the service's schema including the annotations of
+    federation directives. This SDL does not include the additions of the federation spec.
+    """
```
There descriptions were pulled from https://www.apollographql.com/docs/federation/v1/federation-spec/#federation-schema-specification

kzlsakal reviewed (approved):
Awesome contribution @cschiewek!
